### PR TITLE
FIX: Non-native endian bug with largest_connected_component function

### PR DESCRIPTION
**Pull request by KamalakerDadi**
This issue came across with this PR #1409

### CODE REVIEW

File: nilearn/_utils/ndimage.py [hunk 1]
```diff
@@ -17,18 +17,36 @@ def largest_connected_component(volume):
...
+        same manipulation directly on Nifti images.
+
+    Notes
+    -----
+
+    **Handling big-endian in given numpy.ndarray**
+    This function changes the existing byte-ordering information to new byte
+    order, if the given volume has non-native data type. This operation
+    is done inplace to avoid big-endian issues with scipy ndimage module.
+
```
Comment (GaelVaroquaux) [inline nilearn/_utils/ndimage.py]: Is it really done in place? 

Is the input array modified?

Comment (GaelVaroquaux) [inline nilearn/_utils/ndimage.py]: It is indeed

**Comment by GaelVaroquaux** [comment]
LGTM. +1 for merge

File: nilearn/_utils/ndimage.py (hunk 1 above)
Comment (KamalakerDadi) [inline nilearn/_utils/ndimage.py]: May be shoud I reformulate inplace to something ?
Any ideas ?

**Comment by bthirion** [comment]
LGTM  +1 for merge.

File: nilearn/_utils/ndimage.py (hunk 1 above)
Comment (GaelVaroquaux) [inline nilearn/_utils/ndimage.py]: No, it's good.

**Comment by GaelVaroquaux** [comment]
@bthirion gave his +1, so I am merging.

**Pull request merged**